test/e2e/apps/job.go

```diff
@@ -361,6 +368,172 @@ var _ = SIGDescribe("Job", func() {
 			framework.ExpectEqual(pod.Status.Phase, v1.PodFailed)
 		}
 	})
+
+	ginkgo.It("should test the lifecycle of a job", func() {
+		one := int64(1)
+		jobName := "e2e-" + utilrand.String(5)
+		label := map[string]string{"e2e-job-label": jobName}
+		labelSelector := labels.SelectorFromSet(label).String()
+
+		ns := f.Namespace.Name
+		jobClient := f.ClientSet.BatchV1().Jobs(ns)
+
+		w := &cache.ListWatch{
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = labelSelector
+				return jobClient.Watch(context.TODO(), options)
+			},
+		}
+		jobsList, err := jobClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+		framework.ExpectNoError(err, "failed to list Pods")
+
+		ginkgo.By("Creating a suspended job")
+		job := e2ejob.NewTestJob("succeed", jobName, v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
+		job.Labels = label
+		job.Spec.Suspend = pointer.BoolPtr(true)
+		job, err = e2ejob.CreateJob(f.ClientSet, ns, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", ns)
+
+		ginkgo.By("Ensuring pods aren't created for job")
+		framework.ExpectEqual(wait.Poll(framework.Poll, wait.ForeverTestTimeout, func() (bool, error) {
+			pods, err := e2ejob.GetJobPods(f.ClientSet, ns, job.Name)
+			if err != nil {
+				return false, err
+			}
+			return len(pods.Items) > 0, nil
+		}), wait.ErrWaitTimeout)
+
+		ginkgo.By("Checking Job status to observe Suspended state")
+		job, err = e2ejob.GetJob(f.ClientSet, ns, job.Name)
+		framework.ExpectNoError(err, "failed to retrieve latest job object")
+		exists := false
+		for _, c := range job.Status.Conditions {
+			if c.Type == batchv1.JobSuspended {
+				exists = true
+				break
+			}
+		}
+		framework.ExpectEqual(exists, true)
+
+		ginkgo.By("Patching the Job")
+		patchJob, err := json.Marshal(v1.PodTemplateSpec{
```

Comment:
why `PodTemplateSpec`. Shouldn't it be `Job`?

Reply:
The name `patchJob` was a bit confusing. I've renamed it to `payload`. From what I see in the documentation Job is using a [pod template](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-template) within the spec. From the e2e test logs it passes after finding the required label. Have I missed something?